src/native/corehost/corehost.proj

```diff
@@ -18,11 +18,19 @@
   <Target Name="PrepareSingleFileHostWithEmbeddedDacPrereqs">
     <PropertyGroup>
       <SingleFileHostPath>$(CoreCLRArtifactsPath)/corehost/singlefilehost$(ExeSuffix)</SingleFileHostPath>
-      <SingleFileHostSymbolsPath>$(CoreCLRArtifactsPath)/corehost/PDB/singlefilehost$(SymbolsSuffix)</SingleFileHostSymbolsPath>
       <DacPath>$(CoreCLRArtifactsPath)/mscordaccore$(LibSuffix)</DacPath>
       <SingleFileHostDestinationPath>$(DotNetHostBinDir)/singlefilehost$(ExeSuffix)</SingleFileHostDestinationPath>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(TargetOS)' == 'windows'">
+      <!-- Symbols for Windows are in a PDB subdirectory -->
+      <SingleFileHostSymbolsPath>$(CoreCLRArtifactsPath)/corehost/PDB/singlefilehost$(SymbolsSuffix)</SingleFileHostSymbolsPath>
       <SingleFileHostSymbolsDestinationPath>$(DotNetHostBinDir)/PDB/singlefilehost$(SymbolsSuffix)</SingleFileHostSymbolsDestinationPath>
     </PropertyGroup>
+    <PropertyGroup Condition="'$(TargetOS)' != 'windows'">
+      <!-- Symbols for non-Windows are next to the binary -->
+      <SingleFileHostSymbolsPath>$(CoreCLRArtifactsPath)/corehost/singlefilehost$(SymbolsSuffix)</SingleFileHostSymbolsPath>
+      <SingleFileHostSymbolsDestinationPath>$(DotNetHostBinDir)/singlefilehost$(SymbolsSuffix)</SingleFileHostSymbolsDestinationPath>
```

Comment:
I just noticed that https://github.com/dotnet/runtime/blob/a5f3676cc71e176084f0f7f1f6beeecd86fbeafc/eng/native/naming.props#L39-L43 is now a dead code (since https://github.com/dotnet/arcade/commit/99968bd08a1d0a031a557d40557028b05ce8c2fc). We can clean it up.